### ISSUE TITLE
F811: Check TypeVar defaults and stringised type parameters in bases

### ISSUE THREAD

**Issue by Gobot1234** [astral-sh/ruff#5952]
I have code that looks like
```py
from typing import TYPE_CHECKING

if TYPE_CHECKING:
	import some_circular_mod

def bar():
	import some_circular_mod  # F811 Redefinition of unused `some_circular_mod` from line 4
```
I don't think an error should be reported in this case and ruff should check if the module is imported in TYPE_CHECKING before emitting the error.

**Comment by charliermarsh** [comment]
Is `some_circular_mod` being used in type annotations elsewhere?

**Comment by Gobot1234** [comment]
Yeah, sorry missed that from the example

**Comment by charliermarsh** [comment]
Do you mind updating the example to include a usage? For example, this _doesn't_ raise F811, which seems correct:

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    import some_circular_mod

def bar():
    import some_circular_mod

def foo():
    x: some_circular_mod.A = 1
```

**Comment by Gobot1234** [comment]
Sorry again, didn't take the time to check, 
```py
from typing import TYPE_CHECKING
from typing_extensions import TypeVar

if TYPE_CHECKING:
    import some_circular_mod

T = TypeVar("T", default="some_circular_mod.X")  # problematically not checked


def bar():
    import some_circular_mod

class Foo(Generic[T]): pass
class Bar(Foo["some_circular_mod.Y"]): pass  # problematically not checked
```

**Comment by charliermarsh** [comment]
Thanks!

**Comment by ntBre** [comment]
I believe this was fixed in 0.1.12.
